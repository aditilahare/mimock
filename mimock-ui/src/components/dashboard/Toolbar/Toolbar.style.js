import tw from 'tailwind-styled-components';
import { CustomButton } from 'styles/Button';

export const ToolbarContainer = tw.div`
    w-11/12
    flex
    flex-row
    mx-auto
    mt-4
    mb-2
    justify-center
    align-middle
`;

export const ToolbarInnerContainer = tw.div`
    w-5/6
    flex
    flex-row
    mx-auto
    justify-end
    align-middle
    px-2
`;

export const AddMockButton = tw(CustomButton)``;

export const FilterMockButton = tw(CustomButton)``;

export const SelectWrapper = tw.div`
    flex
    flex-col
    mx-2
    my-2
    w-1/6
`;

export const SelectLabel = tw.label`
    text-gray-400
    text-xs
`;

export const SelectComponent = tw.select`
    w-full
    justify-center
    border-2
    rounded-sm
    border-gray-200
    text-teal-500
    outline-none
    mx-2
    my-2
`;

export const SelectOptionComponent = tw.option`
    bg-white
    truncate
    rounded-sm
    outline-none
    text-gray-400
`;
