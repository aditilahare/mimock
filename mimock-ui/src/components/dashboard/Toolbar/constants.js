export const constants = {
	label: {
		add: 'Add Mock',
		sidebar: 'Tools',
		generalView: 'Active',
		archivedView: 'Archived',
		recycleBinView: 'Bin',
	},
	name: {
		generalView: 'ACTIVE',
		archivedView: 'ARCHIVED',
		recycleBinView: 'DELETED',
	},
};
