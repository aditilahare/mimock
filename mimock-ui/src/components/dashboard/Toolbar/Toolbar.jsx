import React, { useEffect, useState } from 'react';
import {
	ToolbarContainer,
	ToolbarInnerContainer,
	AddMockButton,
	addMocksButtonAdditionalStyles,
	MoreToolsMockButton,
	moreToolsMockButtonAdditonalStyles,
	ViewMocksWrapper,
	ViewMocksInnerWrapper,
	GeneralMocksViewButton,
	ArchivedMocksViewButton,
	RecycleBinViewButton,
	generalMocksViewButtonAdditionalStyles,
	archivedMocksViewButtonAdditionalStyles,
	recycleBinViewButtonAdditionalStyles,
} from './Toolbar.style';
import { constants } from './constants';
import PropTypes from 'prop-types';
import { FaCogs, FaPlusSquare } from 'react-icons/fa';
import { MdDone, MdArchive, MdDelete } from 'react-icons/md';
import { mockManagementConstants } from 'constants/globalConstants';
import { useNavigate } from 'react-router-dom';
import { routes } from 'constants/routes';
import { getUserDetails } from 'utils/jwtUtils';

function Toolbar({
	handleSidebarBtnClick,
	handleMockListsView,
	mocksListView,
}) {
	const activeClassList = [
		'bg-teal-500',
		'text-white',
		'text-semibold',
		'hover:bg-gray-500',
		'hover-bg-opacity-5',
		'hover:text-teal-500',
	];
	const navigate = useNavigate();

	const [hideAddMockBtn, setHideAddMockBtn] = useState(false);

	useEffect(() => {
		try {
			const userDetails = getUserDetails();
			const isReadOnlyUser =
				userDetails && userDetails.userRole === 'ROLE_VIEWER';
			setHideAddMockBtn(isReadOnlyUser);
		} catch (e) {
			console.log(e);
			setHideAddMockBtn(false);
		}
	});

	useEffect(() => {
		if (mocksListView == mockManagementConstants.DEFAULT_STATUS) {
			const activeMockBtn = document.getElementById('active-view-btn');
			const archivedMockBtn = document.getElementById('archived-view-btn');
			const deletedMockBtn = document.getElementById('deleted-view-btn');
			activeMockBtn.classList.remove(...activeClassList);
			archivedMockBtn.classList.remove(...activeClassList);
			deletedMockBtn.classList.remove(...activeClassList);
		}
	}, [mocksListView]);

	const handleAddMockBtnClick = () => {
		navigate(routes.manageRoutes.addMock.path);
	};

	const handleViewMocksBtnClick = (e) => {
		const targetElement = e.target;
		const parentElement = e.target.parentElement;
		let viewName = '';
		const activeMockBtn = document.getElementById('active-view-btn');
		const archivedMockBtn = document.getElementById('archived-view-btn');
		const deletedMockBtn = document.getElementById('deleted-view-btn');
		activeMockBtn.classList.remove(...activeClassList);
		archivedMockBtn.classList.remove(...activeClassList);
		deletedMockBtn.classList.remove(...activeClassList);
		if (
			targetElement != undefined &&
			parentElement != undefined &&
			targetElement.getAttribute('name') != undefined
		) {
			viewName = targetElement.getAttribute('name');
		} else {
			viewName = parentElement.getAttribute('name');
		}
		switch (viewName) {
			case 'ACTIVE':
				activeMockBtn.classList.add(...activeClassList);
				break;
			case 'ARCHIVED':
				archivedMockBtn.classList.add(...activeClassList);
				break;
			case 'DELETED':
				deletedMockBtn.classList.add(...activeClassList);
				break;
		}
		handleMockListsView(viewName);
	};

	return (
		<>
			<ToolbarContainer data-testid='toolbar-section'>
				<MoreToolsMockButton
					dataTestid='more-tools-btn'
					background='bg-white'
					label={constants.label.sidebar}
					color='text-gray-500'
					icon={<FaCogs />}
					additionalStyles={moreToolsMockButtonAdditonalStyles}
					onclickHandler={handleSidebarBtnClick}
				></MoreToolsMockButton>
				<ToolbarInnerContainer>
					<ViewMocksWrapper>
						<ViewMocksInnerWrapper role='group'>
							<GeneralMocksViewButton
								dataTestid='active-view-btn'
								id='active-view-btn'
								background='bg-white'
								label={constants.label.generalView}
								color='text-gray-500'
								icon={<MdDone name={constants.name.generalView} />}
								additionalStyles={generalMocksViewButtonAdditionalStyles}
								name={constants.name.generalView}
								onclickHandler={handleViewMocksBtnClick}
							></GeneralMocksViewButton>
							<ArchivedMocksViewButton
								dataTestid='archived-view-btn'
								id='archived-view-btn'
								background='bg-white'
								label={constants.label.archivedView}
								color='text-gray-500'
								icon={<MdArchive name={constants.name.archivedView} />}
								additionalStyles={archivedMocksViewButtonAdditionalStyles}
								name={constants.name.archivedView}
								onclickHandler={handleViewMocksBtnClick}
							></ArchivedMocksViewButton>
							<RecycleBinViewButton
								dataTestid='deleted-view-btn'
								id='deleted-view-btn'
								background='bg-white'
								label={constants.label.recycleBinView}
								color='text-gray-500'
								icon={<MdDelete name={constants.name.recycleBinView} />}
								additionalStyles={recycleBinViewButtonAdditionalStyles}
								name={constants.name.recycleBinView}
								onclickHandler={handleViewMocksBtnClick}
							></RecycleBinViewButton>
						</ViewMocksInnerWrapper>
					</ViewMocksWrapper>
					<If condition={!hideAddMockBtn}>
						<AddMockButton
							dataTestid='add-btn'
							background='bg-teal-500'
							label={constants.label.add}
							color='text-white'
							icon={<FaPlusSquare />}
							additionalStyles={addMocksButtonAdditionalStyles}
							onclickHandler={handleAddMockBtnClick}
						></AddMockButton>
					</If>
				</ToolbarInnerContainer>
			</ToolbarContainer>
			<hr />
		</>
	);
}

Toolbar.propTypes = {
	handleSidebarBtnClick: PropTypes.func.isRequired,
	handleMockListsView: PropTypes.func.isRequired,
	mocksListView: PropTypes.string.isRequired,
};

export default Toolbar;
