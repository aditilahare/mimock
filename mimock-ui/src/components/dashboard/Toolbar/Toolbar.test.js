import React from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import Toolbar from './Toolbar';

const mockedNavigation = jest.fn(() => {});
jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockedNavigation,
}));

describe('Toolbar', () => {
	it('should render toolbar component', async () => {
		const tree = await render(<Toolbar />);

		const { container, getByTestId } = tree;

		expect(getByTestId('toolbar-container')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should navigate to add mock page on clicking add CTA', async () => {
		const tree = await render(<Toolbar />);

		const { container, getByTestId } = tree;

		expect(getByTestId('toolbar-container')).toBeInTheDocument();

		await act(async () => {
			fireEvent.click(getByTestId('add-mock-btn'));
		});

		expect(mockedNavigation).toHaveBeenCalledTimes(1);
		expect(mockedNavigation).toHaveBeenCalledWith(
			'/mimock-ui/manage/mocks/add-mock'
		);

		expect(container).toMatchSnapshot();
	});
});
