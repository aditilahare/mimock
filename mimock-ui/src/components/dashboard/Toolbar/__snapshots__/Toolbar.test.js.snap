// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should navigate to add mock page on clicking add CTA 1`] = `
<div>
  <div
    class="w-11/12 flex flex-row mx-auto mt-4 mb-4 justify-center align-middle px-2 md:items-center"
    data-testid="toolbar-section"
  >
    <button
      class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 h-10 mx-4 my-2 rounded-sm text-sm border-2 border-gray-400 hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out md:my-2 text-gray-500 w-fit bg-white"
      data-testid="more-tools-btn"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512.1 191l-8.2 14.3c-3 5.3-9.4 7.5-15.1 5.4-11.8-4.4-22.6-10.7-32.1-18.6-4.6-3.8-5.8-10.5-2.8-15.7l8.2-14.3c-6.9-8-12.3-17.3-15.9-27.4h-16.5c-6 0-11.2-4.3-12.2-10.3-2-12-2.1-24.6 0-37.1 1-6 6.2-10.4 12.2-10.4h16.5c3.6-10.1 9-19.4 15.9-27.4l-8.2-14.3c-3-5.2-1.9-11.9 2.8-15.7 9.5-7.9 20.4-14.2 32.1-18.6 5.7-2.1 12.1.1 15.1 5.4l8.2 14.3c10.5-1.9 21.2-1.9 31.7 0L552 6.3c3-5.3 9.4-7.5 15.1-5.4 11.8 4.4 22.6 10.7 32.1 18.6 4.6 3.8 5.8 10.5 2.8 15.7l-8.2 14.3c6.9 8 12.3 17.3 15.9 27.4h16.5c6 0 11.2 4.3 12.2 10.3 2 12 2.1 24.6 0 37.1-1 6-6.2 10.4-12.2 10.4h-16.5c-3.6 10.1-9 19.4-15.9 27.4l8.2 14.3c3 5.2 1.9 11.9-2.8 15.7-9.5 7.9-20.4 14.2-32.1 18.6-5.7 2.1-12.1-.1-15.1-5.4l-8.2-14.3c-10.4 1.9-21.2 1.9-31.7 0zm-10.5-58.8c38.5 29.6 82.4-14.3 52.8-52.8-38.5-29.7-82.4 14.3-52.8 52.8zM386.3 286.1l33.7 16.8c10.1 5.8 14.5 18.1 10.5 29.1-8.9 24.2-26.4 46.4-42.6 65.8-7.4 8.9-20.2 11.1-30.3 5.3l-29.1-16.8c-16 13.7-34.6 24.6-54.9 31.7v33.6c0 11.6-8.3 21.6-19.7 23.6-24.6 4.2-50.4 4.4-75.9 0-11.5-2-20-11.9-20-23.6V418c-20.3-7.2-38.9-18-54.9-31.7L74 403c-10 5.8-22.9 3.6-30.3-5.3-16.2-19.4-33.3-41.6-42.2-65.7-4-10.9.4-23.2 10.5-29.1l33.3-16.8c-3.9-20.9-3.9-42.4 0-63.4L12 205.8c-10.1-5.8-14.6-18.1-10.5-29 8.9-24.2 26-46.4 42.2-65.8 7.4-8.9 20.2-11.1 30.3-5.3l29.1 16.8c16-13.7 34.6-24.6 54.9-31.7V57.1c0-11.5 8.2-21.5 19.6-23.5 24.6-4.2 50.5-4.4 76-.1 11.5 2 20 11.9 20 23.6v33.6c20.3 7.2 38.9 18 54.9 31.7l29.1-16.8c10-5.8 22.9-3.6 30.3 5.3 16.2 19.4 33.2 41.6 42.1 65.8 4 10.9.1 23.2-10 29.1l-33.7 16.8c3.9 21 3.9 42.5 0 63.5zm-117.6 21.1c59.2-77-28.7-164.9-105.7-105.7-59.2 77 28.7 164.9 105.7 105.7zm243.4 182.7l-8.2 14.3c-3 5.3-9.4 7.5-15.1 5.4-11.8-4.4-22.6-10.7-32.1-18.6-4.6-3.8-5.8-10.5-2.8-15.7l8.2-14.3c-6.9-8-12.3-17.3-15.9-27.4h-16.5c-6 0-11.2-4.3-12.2-10.3-2-12-2.1-24.6 0-37.1 1-6 6.2-10.4 12.2-10.4h16.5c3.6-10.1 9-19.4 15.9-27.4l-8.2-14.3c-3-5.2-1.9-11.9 2.8-15.7 9.5-7.9 20.4-14.2 32.1-18.6 5.7-2.1 12.1.1 15.1 5.4l8.2 14.3c10.5-1.9 21.2-1.9 31.7 0l8.2-14.3c3-5.3 9.4-7.5 15.1-5.4 11.8 4.4 22.6 10.7 32.1 18.6 4.6 3.8 5.8 10.5 2.8 15.7l-8.2 14.3c6.9 8 12.3 17.3 15.9 27.4h16.5c6 0 11.2 4.3 12.2 10.3 2 12 2.1 24.6 0 37.1-1 6-6.2 10.4-12.2 10.4h-16.5c-3.6 10.1-9 19.4-15.9 27.4l8.2 14.3c3 5.2 1.9 11.9-2.8 15.7-9.5 7.9-20.4 14.2-32.1 18.6-5.7 2.1-12.1-.1-15.1-5.4l-8.2-14.3c-10.4 1.9-21.2 1.9-31.7 0zM501.6 431c38.5 29.6 82.4-14.3 52.8-52.8-38.5-29.6-82.4 14.3-52.8 52.8z"
        />
      </svg>
      Tools
    </button>
    <div
      class="w-5/6 flex xl:flex-row md:flex-col mx-auto justify-end align-middle px-2"
    >
      <div
        class="flex flex-row items-center justify-center md:my-2"
      >
        <div
          class="inline-flex"
          role="group"
        >
          <button
            class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 h-10 rounded-l-sm border-2 border-gray-400 rounded-r-none text-sm leading-tight hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out text-gray-500 w-fit bg-white"
            data-testid="active-view-btn"
            id="active-view-btn"
            name="ACTIVE"
          >
            <svg
              fill="currentColor"
              height="1em"
              name="ACTIVE"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            Active
          </button>
          <button
            class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 h-10 rounded-none text-sm border-t-2 border-b-2 border-gray-400 hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out text-gray-500 w-fit bg-white"
            data-testid="archived-view-btn"
            id="archived-view-btn"
            name="ARCHIVED"
          >
            <svg
              fill="currentColor"
              height="1em"
              name="ARCHIVED"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20.54 5.23l-1.39-1.68C18.88 3.21 18.47 3 18 3H6c-.47 0-.88.21-1.16.55L3.46 5.23C3.17 5.57 3 6.02 3 6.5V19c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V6.5c0-.48-.17-.93-.46-1.27zM12 17.5L6.5 12H10v-2h4v2h3.5L12 17.5zM5.12 5l.81-1h12l.94 1H5.12z"
              />
            </svg>
            Archived
          </button>
          <button
            class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 h-10 rounded-r-sm border-2 border-gray-400 rounded-l-none text-sm leading-tight hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out text-gray-500 w-fit bg-white"
            data-testid="deleted-view-btn"
            id="deleted-view-btn"
            name="DELETED"
          >
            <svg
              fill="currentColor"
              height="1em"
              name="DELETED"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            Bin
          </button>
        </div>
      </div>
      <button
        class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 h-10 xl:ml-8 xl:mr-2 xl:my-2 px-1 py-1 rounded-sm text-sm focus:outline-none focus:ring-0 transition duration-150 ease-in-out xl:w-1/6 md:w-5/6 md:my-2 md:mx-auto text-white w-fit bg-teal-500"
        data-testid="add-btn"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
          />
        </svg>
        Add Mock
      </button>
    </div>
  </div>
  <hr />
</div>
`;

exports[`Toolbar should render toolbar component 1`] = `
<div>
  <div
    class="w-11/12 flex flex-row mx-auto mt-4 mb-4 justify-center align-middle px-2 md:items-center"
    data-testid="toolbar-section"
  >
    <button
      class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 h-10 mx-4 my-2 rounded-sm text-sm border-2 border-gray-400 hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out md:my-2 text-gray-500 w-fit bg-white"
      data-testid="more-tools-btn"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512.1 191l-8.2 14.3c-3 5.3-9.4 7.5-15.1 5.4-11.8-4.4-22.6-10.7-32.1-18.6-4.6-3.8-5.8-10.5-2.8-15.7l8.2-14.3c-6.9-8-12.3-17.3-15.9-27.4h-16.5c-6 0-11.2-4.3-12.2-10.3-2-12-2.1-24.6 0-37.1 1-6 6.2-10.4 12.2-10.4h16.5c3.6-10.1 9-19.4 15.9-27.4l-8.2-14.3c-3-5.2-1.9-11.9 2.8-15.7 9.5-7.9 20.4-14.2 32.1-18.6 5.7-2.1 12.1.1 15.1 5.4l8.2 14.3c10.5-1.9 21.2-1.9 31.7 0L552 6.3c3-5.3 9.4-7.5 15.1-5.4 11.8 4.4 22.6 10.7 32.1 18.6 4.6 3.8 5.8 10.5 2.8 15.7l-8.2 14.3c6.9 8 12.3 17.3 15.9 27.4h16.5c6 0 11.2 4.3 12.2 10.3 2 12 2.1 24.6 0 37.1-1 6-6.2 10.4-12.2 10.4h-16.5c-3.6 10.1-9 19.4-15.9 27.4l8.2 14.3c3 5.2 1.9 11.9-2.8 15.7-9.5 7.9-20.4 14.2-32.1 18.6-5.7 2.1-12.1-.1-15.1-5.4l-8.2-14.3c-10.4 1.9-21.2 1.9-31.7 0zm-10.5-58.8c38.5 29.6 82.4-14.3 52.8-52.8-38.5-29.7-82.4 14.3-52.8 52.8zM386.3 286.1l33.7 16.8c10.1 5.8 14.5 18.1 10.5 29.1-8.9 24.2-26.4 46.4-42.6 65.8-7.4 8.9-20.2 11.1-30.3 5.3l-29.1-16.8c-16 13.7-34.6 24.6-54.9 31.7v33.6c0 11.6-8.3 21.6-19.7 23.6-24.6 4.2-50.4 4.4-75.9 0-11.5-2-20-11.9-20-23.6V418c-20.3-7.2-38.9-18-54.9-31.7L74 403c-10 5.8-22.9 3.6-30.3-5.3-16.2-19.4-33.3-41.6-42.2-65.7-4-10.9.4-23.2 10.5-29.1l33.3-16.8c-3.9-20.9-3.9-42.4 0-63.4L12 205.8c-10.1-5.8-14.6-18.1-10.5-29 8.9-24.2 26-46.4 42.2-65.8 7.4-8.9 20.2-11.1 30.3-5.3l29.1 16.8c16-13.7 34.6-24.6 54.9-31.7V57.1c0-11.5 8.2-21.5 19.6-23.5 24.6-4.2 50.5-4.4 76-.1 11.5 2 20 11.9 20 23.6v33.6c20.3 7.2 38.9 18 54.9 31.7l29.1-16.8c10-5.8 22.9-3.6 30.3 5.3 16.2 19.4 33.2 41.6 42.1 65.8 4 10.9.1 23.2-10 29.1l-33.7 16.8c3.9 21 3.9 42.5 0 63.5zm-117.6 21.1c59.2-77-28.7-164.9-105.7-105.7-59.2 77 28.7 164.9 105.7 105.7zm243.4 182.7l-8.2 14.3c-3 5.3-9.4 7.5-15.1 5.4-11.8-4.4-22.6-10.7-32.1-18.6-4.6-3.8-5.8-10.5-2.8-15.7l8.2-14.3c-6.9-8-12.3-17.3-15.9-27.4h-16.5c-6 0-11.2-4.3-12.2-10.3-2-12-2.1-24.6 0-37.1 1-6 6.2-10.4 12.2-10.4h16.5c3.6-10.1 9-19.4 15.9-27.4l-8.2-14.3c-3-5.2-1.9-11.9 2.8-15.7 9.5-7.9 20.4-14.2 32.1-18.6 5.7-2.1 12.1.1 15.1 5.4l8.2 14.3c10.5-1.9 21.2-1.9 31.7 0l8.2-14.3c3-5.3 9.4-7.5 15.1-5.4 11.8 4.4 22.6 10.7 32.1 18.6 4.6 3.8 5.8 10.5 2.8 15.7l-8.2 14.3c6.9 8 12.3 17.3 15.9 27.4h16.5c6 0 11.2 4.3 12.2 10.3 2 12 2.1 24.6 0 37.1-1 6-6.2 10.4-12.2 10.4h-16.5c-3.6 10.1-9 19.4-15.9 27.4l8.2 14.3c3 5.2 1.9 11.9-2.8 15.7-9.5 7.9-20.4 14.2-32.1 18.6-5.7 2.1-12.1-.1-15.1-5.4l-8.2-14.3c-10.4 1.9-21.2 1.9-31.7 0zM501.6 431c38.5 29.6 82.4-14.3 52.8-52.8-38.5-29.6-82.4 14.3-52.8 52.8z"
        />
      </svg>
      Tools
    </button>
    <div
      class="w-5/6 flex xl:flex-row md:flex-col mx-auto justify-end align-middle px-2"
    >
      <div
        class="flex flex-row items-center justify-center md:my-2"
      >
        <div
          class="inline-flex"
          role="group"
        >
          <button
            class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 h-10 rounded-l-sm border-2 border-gray-400 rounded-r-none text-sm leading-tight hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out text-gray-500 w-fit bg-white"
            data-testid="active-view-btn"
            id="active-view-btn"
            name="ACTIVE"
          >
            <svg
              fill="currentColor"
              height="1em"
              name="ACTIVE"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            Active
          </button>
          <button
            class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 h-10 rounded-none text-sm border-t-2 border-b-2 border-gray-400 hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out text-gray-500 w-fit bg-white"
            data-testid="archived-view-btn"
            id="archived-view-btn"
            name="ARCHIVED"
          >
            <svg
              fill="currentColor"
              height="1em"
              name="ARCHIVED"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20.54 5.23l-1.39-1.68C18.88 3.21 18.47 3 18 3H6c-.47 0-.88.21-1.16.55L3.46 5.23C3.17 5.57 3 6.02 3 6.5V19c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V6.5c0-.48-.17-.93-.46-1.27zM12 17.5L6.5 12H10v-2h4v2h3.5L12 17.5zM5.12 5l.81-1h12l.94 1H5.12z"
              />
            </svg>
            Archived
          </button>
          <button
            class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 h-10 rounded-r-sm border-2 border-gray-400 rounded-l-none text-sm leading-tight hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out text-gray-500 w-fit bg-white"
            data-testid="deleted-view-btn"
            id="deleted-view-btn"
            name="DELETED"
          >
            <svg
              fill="currentColor"
              height="1em"
              name="DELETED"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            Bin
          </button>
        </div>
      </div>
      <button
        class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 h-10 xl:ml-8 xl:mr-2 xl:my-2 px-1 py-1 rounded-sm text-sm focus:outline-none focus:ring-0 transition duration-150 ease-in-out xl:w-1/6 md:w-5/6 md:my-2 md:mx-auto text-white w-fit bg-teal-500"
        data-testid="add-btn"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
          />
        </svg>
        Add Mock
      </button>
    </div>
  </div>
  <hr />
</div>
`;
